FunctionsHW12.cpp

```diff
@@ -0,0 +1,97 @@
+#include "FunctionsHW12.h"
+
+void bubbleSort(int arr[], int size) {
+    for (int i = 0; i < size; i++) {
+        for (int j = 0; j < size - 1 - i; j++) {
+            if (arr[j] > arr[j + 1]) {
+                int temp = arr[j];
+                arr[j] = arr[j + 1];
+                arr[j + 1] = temp;
+            }
+        }
+    }
+}
+
+void sort(int arr[ROWS][COLUMNS], SortingDirection sd) {
+    switch (sd) {
+    case SortingDirection::ByRows:
+        for (int r = 0; r < ROWS; r++) {
+            quickSortByRows(arr, r, 0, COLUMNS - 1);
+        }
+        break;
+    case SortingDirection::ByColumns:
+        for (int c = 0; c < COLUMNS; c++) {
+            quickSortByColumns(arr, c, 0, ROWS - 1);
+        }
+        break;
```

Comment:
it works, but in this task, you slould use this function `void quickSort(int arr[], int startIdx, int endIdx) ` for sorting rows and columns, but you should pass the right arguments